pithos/plugins/notification_icon.py

```diff
@@ -13,56 +13,63 @@
 # with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 import sys
+import logging
 import gi
 
 from gi.repository import GObject, Gdk, Gtk
 
 from pithos.plugin import PithosPlugin
 
-# Use appindicator if installed
-try:
-    gi.require_version('AppIndicator3', '0.1')
-    from gi.repository import AppIndicator3 as AppIndicator
-    indicator_capable = True
-except (ImportError, ValueError):
-    indicator_capable = False
-
-
-def backend_is_supported(window):
-    if sys.platform in ('win32', 'darwin'):
-        return True
-    display = window.props.screen.get_display()
-    return type(display).__name__.endswith('X11Display')
-
 
 class PithosNotificationIcon(PithosPlugin):
     preference = 'show_icon'
     description = 'Adds pithos icon to system tray'
 
+    indicator_capable = False
+
     def on_prepare(self):
-        if indicator_capable:
-            self.ind = AppIndicator.Indicator.new("pithos-tray-icon",
-                                                  "pithos-tray-icon",
-                                                  AppIndicator.IndicatorCategory.APPLICATION_STATUS)
-            # FIXME: AppIndicator might be falling back to XEmbed
-        elif not backend_is_supported(self.window):
-            self.prepare_complete(error='Notification icon requires X11 or AppIndicator')
-        else:
-            self.prepare_complete()
+        try:
+            gi.require_version('AppIndicator3', '0.1')
+            from gi.repository import AppIndicator3 as AppIndicator
+            self.indicator_capable = True
+        except (ImportError, ValueError):
+            self.indicator_capable = False
+
+        if self.indicator_capable:
+            try:
+                self.ind = AppIndicator.Indicator.new(
```

Comment:
I'm not sure? I just wanted to cover our bases since the tray icon doesn't work with Wayland. I'm don't even know how to test AppIndicator against Wayland? I suppose I can install  AppIndicator in my Ubuntu GNOME VM and try it out in the Wayland session with the AppIndicator Shell extension?